src/algorithms/calorimetry/SimCalorimeterHitProcessor.cc

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2025 Minho Kim, Sylvester Joosten, Derek Anderson
+
+#include "SimCalorimeterHitProcessor.h"
+
+#include <DD4hep/Detector.h>
+#include <DD4hep/IDDescriptor.h>
+#include <DD4hep/Readout.h>
+#include <DD4hep/config.h>
+#include <DDSegmentation/BitFieldCoder.h>
+#include <Evaluator/DD4hepUnits.h>
+#include <algorithms/service.h>
+#include <fmt/core.h>
+#include <podio/RelationRange.h>
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <gsl/pointers>
+#include <limits>
+#include <stdexcept>
+#include <string>
+#include <map>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "algorithms/calorimetry/SimCalorimeterHitProcessorConfig.h"
+#include "services/evaluator/EvaluatorSvc.h"
+
+using namespace dd4hep;
+
+namespace eicrecon {
+
+void SimCalorimeterHitProcessor::init() {
+
+  // readout checks
+  if (m_cfg.readout.empty()) {
+    error("readoutClass is not provided, it is needed to know the fields in readout ids");
+    throw std::runtime_error("readoutClass is not provided");
+  }
+
+  // get decoders
+  try {
+    m_id_spec = m_geo.detector()->readout(m_cfg.readout).idSpec();
+  } catch (...) {
+    debug("Failed to load ID decoder for {}", m_cfg.readout);
+    throw std::runtime_error(fmt::format("Failed to load ID decoder for {}", m_cfg.readout));
+  }
+
+  uint64_t id_inverse_mask = 0;
+  // get id_mask for adding up hits that have the same z-segmentation
```

Comment:
Remove references to "z-segmentation" in comments and code. This class is supposed to be generic. so this should refer generically to "dimensions that are merged over" or something.